### PR TITLE
Added more player-related events

### PR DESCRIPTION
**Pull request by ebo2022**

### CODE REVIEW

File: forge/src/main/java/gg/moonflower/pollen/core/forge/PollenCommonForgeEvents.java [hunk 1]
```diff
@@ -171,6 +178,19 @@ public static void onEvent(net.minecraftforge.event.world.ExplosionEvent.Detonat
         ExplosionEvents.DETONATE.invoker().detonate(event.getWorld(), event.getExplosion(), event.getAffectedEntities());
     }
 
+    @SubscribeEvent
+    public static void onEvent(net.minecraftforge.event.world.SaplingGrowTreeEvent event) {
+        InteractionResult result = TreeGrowingEvent.EVENT.invoker().interaction(event.getWorld(), event.getRand(), event.getPos());
+        switch (result) {
```
Comment (Ocelot5836) [inline forge/src/main/java/gg/moonflower/pollen/core/forge/PollenCommonForgeEvents.java]: This switch statement needs to have breaks or else all branches will fall back to `event.setResult(Event.Result.DENY);`

**Review by Ocelot5836 — CHANGES_REQUESTED** [review]
The Forge `SaplingGrowTreeEvent` needs to be fixed. Otherwise, these events look like good additions.

**Comment by Ocelot5836** [comment]
I would also like for you to merge into `next/1.16.x` since that is the branch for the next major/minor release.

File: forge/src/main/java/gg/moonflower/pollen/core/forge/PollenCommonForgeEvents.java (hunk 1 above)
Comment (ebo2022) [reply forge/src/main/java/gg/moonflower/pollen/core/forge/PollenCommonForgeEvents.java]: This issue should hopefully be fixed in the latest commit, I also removed the duplicate case statement

**Review by Ocelot5836 — APPROVED** [review]
Looks good to me. I'm waiting on Jackson to add a cla before this gets merged.

**Comment by ebo2022** [comment]
Alright, sounds good

**Comment by ebo2022** [comment]
Closing this for now since I no longer have a need for many of these events, and the code in general is very messy. I will probably open some new PRs once more progress is made on 1.19

**Comment by ebo2022** [comment]
Re-opening this. Going to clean up code so it's easier to implement this time around hopefully

**Pull request closed without merging**